dhis-2/dhis-api/src/main/java/org/hisp/dhis/common/ValueTypeOptions.java

```diff
@@ -0,0 +1,69 @@
+package org.hisp.dhis.common;
+
+/*
+ * Copyright (c) 2004-2021, University of Oslo
+ * All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions are met:
+ * Redistributions of source code must retain the above copyright notice, this
+ * list of conditions and the following disclaimer.
+ *
+ * Redistributions in binary form must reproduce the above copyright notice,
+ * this list of conditions and the following disclaimer in the documentation
+ * and/or other materials provided with the distribution.
+ * Neither the name of the HISP project nor the names of its contributors may
+ * be used to endorse or promote products derived from this software without
+ * specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND
+ * ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED
+ * WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+ * DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR
+ * ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+ * (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+ * LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON
+ * ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+ * SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+import com.fasterxml.jackson.annotation.JsonInclude;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.fasterxml.jackson.annotation.JsonSubTypes;
+import com.fasterxml.jackson.annotation.JsonTypeInfo;
+import com.fasterxml.jackson.dataformat.xml.annotation.JacksonXmlRootElement;
+
+import java.io.Serializable;
+
+/**
+ * @author Morten Svanæs <msvanaes@dhis2.org>
+ */
+@JacksonXmlRootElement( localName = "valuetypeoptions", namespace = DxfNamespaces.DXF_2_0 )
+@JsonInclude()
+@JsonTypeInfo( use = JsonTypeInfo.Id.NAME, property = "type" )
+@JsonSubTypes( {
+    @JsonSubTypes.Type( value = FileTypeValueOptions.class, name = "File" ),
+    @JsonSubTypes.Type( value = DigitsValueTypeOptions.class, name = "Digits" )
+} )
```

Comment:
Thinking about extensibility here, do we want or expect 3rd party to be able to add new types? Would this always require to add the new type to this list here or are there options to do the same thing that do not require to modify this file (as this might be "external code" to a 3rd party) ?

Reply:
It is good to think about extensibility in general, but I think in this case we will very likely not see "plugins" in the DHIS 2 core source code.

Reply:
When you were talking about this in the last weekly I got the impression we wanted to add this more flexible type because of various use case based features which I understood as being implemented on top of the core what is why I asked.

Reply:
We probably have to make some extra custom layer on top of the Jackson's subtype system to accommodate a 3rd party extensible system for this. This would be nice, but too much work to include in this first version IMO.